pkg/apply/desiredset.go

```diff
@@ -160,13 +168,47 @@ func (o desiredSet) WithCacheTypes(igs ...InformerGetter) Apply {
 	}
 
 	for _, ig := range igs {
-		pruneTypes[ig.GroupVersionKind()] = ig.Informer()
+		informer := ig.Informer()
+		if err := AddHashIndexer(informer.GetIndexer()); err != nil {
+			// Ignore repeatedly adding the same indexer for different types
+			if !errors.Is(err, indexerAlreadyExistsErr) {
+				logrus.Warnf("Problem adding hash indexer to informer [%s]: %v", ig.GroupVersionKind().Kind, err)
+			}
+		}
+		pruneTypes[ig.GroupVersionKind()] = informer
 	}
 
 	o.pruneTypes = pruneTypes
 	return o
 }
 
+// AddHashIndexer adds an indexer to each provided informer that indexes objects by the hash annotation value
```

Comment:
Please expand this description with the reasons why a developer should add this option and ideally a usage example.

Reply:
Updated, there was no reason to export this function, thanks!